### ISSUE TITLE
the evaluate command is not working , everything else is ??? See below of the console 

### ISSUE THREAD

**Issue by gbortz27** [rstudio/keras#239]
36480/48000 [=====================>........] - ETA: 0s - loss: 0.0526 - acc: 0.9856
38144/48000 [======================>.......] - ETA: 0s - loss: 0.0525 - acc: 0.9857
39552/48000 [=======================>......] - ETA: 0s - loss: 0.0526 - acc: 0.9855
40832/48000 [========================>.....] - ETA: 0s - loss: 0.0531 - acc: 0.9854
42240/48000 [=========================>....] - ETA: 0s - loss: 0.0526 - acc: 0.9855
43520/48000 [==========================>...] - ETA: 0s - loss: 0.0522 - acc: 0.9856
44672/48000 [==========================>...] - ETA: 0s - loss: 0.0514 - acc: 0.9858
46208/48000 [===========================>..] - ETA: 0s - loss: 0.0524 - acc: 0.9856
47488/48000 [============================>.] - ETA: 0s - loss: 0.0524 - acc: 0.9856
48000/48000 [==============================] - 2s 50us/step - loss: 0.0523 - acc: 0.9856 - val_loss: 0.1065 - val_acc: 0.9808
> plot(history)
> model %>% evaluate(x_test, y_test)
Error in function_list[[k]](value) : could not find function "evaluate"

> evaluate(model,x_test, y_test)
Error: could not find function "evaluate"
> model %>% evaluate(x_test, y_test)
Error in function_list[[k]](value) : could not find function "evaluate"
> model %>% predict_classes(x_test)
   [1] 7 2 1 0 4 1 4 9 5 9 0 6 9 0 1 5 9 7 3 4 9 6 6 5 4 0 7 4 0 1 3 1 3 4 7 2 7 1 2 1 1
  [42] 7 4 2 3 5 1 2 4 4 6 3 5 5 6 0 4 1 9 5 7 8 9 3 7 4 6 4 3 0 7 0 2 9 1 7 3 2 9 7 7 6
  [83] 2 7 8 4 7 3 6 1 3 6 9 3 1 4 1 7 6 9 6 0 5 4 9 9 2 1 9 4 8 7 3 9 7 4 4 4 9 2 5 4 7

**Comment by jjallaire** [comment]
You should make sure you are running the most up to date versions of the tensorflow and keras packages from CRAN (evaluate is an S3 method that is provided by the tensorflow package, perhaps you have an older version of it). 

If updating keras from CRAN doesn't solve the problem, please provide your `sessionInfo()` so I can see what versions of various packages you are running.

**Comment by gbortz27** [comment]
THAT WAS A VERY  FAST RESPONSE !!

I INSTALLED THE keras yesterday and reticulate today

Is it perhaps my R version ?


> sessionInfo()
R version 3.3.2 (2016-10-31)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows >= 8 x64 (build 9200)

locale:
[1] LC_COLLATE=English_South Africa.1252  LC_CTYPE=English_South Africa.1252   
[3] LC_MONETARY=English_South Africa.1252 LC_NUMERIC=C                         
[5] LC_TIME=English_South Africa.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] reticulate_1.3.1 keras_2.0.8.9000

loaded via a namespace (and not attached):
 [1] Rcpp_0.12.12          digest_0.6.12         plyr_1.8.4           
 [4] grid_3.3.2            R6_2.2.2              jsonlite_1.5         
 [7] gtable_0.2.0          magrittr_1.5          evaluate_0.10.1      
[10] scales_0.4.1          ggplot2_2.2.1         stringi_1.1.5        
[13] rlang_0.1.1           tfruns_1.0            reshape2_1.4.2       
[16] lazyeval_0.2.0        whisker_0.3-2         labeling_0.3         
[19] tools_3.3.2           stringr_1.2.0         munsell_0.4.3        
[22] colorspace_1.3-2      base64enc_0.1-3       tensorflow_1.4.0.9003
[25] tibble_1.3.3         


> 


Sent from Mail for Windows 10

**Comment by jjallaire** [comment]
You are running a much older version of Keras than the one on CRAN (2.1.2)

**Comment by gbortz27** [comment]
I just ran the install keras again and it worked once I had the new keras package.  Thanks 😊

While I have you here I want to ask some questions about the Nvidia  settings as I am confused about some of the requirements. I would like to take advantage of GPUs

Ie.

I see that my card ; Nvidia GTX 1050 ti has a version 9 of CUDA driver, and therefore must I use Cuda Toolkit version 9. I cant find version 8 as per the instruction.

With CUDA Version 9 must I still use cuDNN v6.0 and not 7 ..see below

Not sure how to set the path variables mentioned in the text below. The NVIDIA notes are very complex to understand so any assistance would be appreciated.

“If you are installing TensorFlow with GPU support using one of the mechanisms described in this guide, then the following NVIDIA software must be installed on your system:
• CUDA® Toolkit 8.0. For details, see NVIDIA's documentation Ensure that you append the relevant Cuda pathnames to the %PATH% environment variable as described in the NVIDIA documentation.
• The NVIDIA drivers associated with CUDA Toolkit 8.0.
• cuDNN v6.0. For details, see NVIDIA's documentation. Note that cuDNN is typically installed in a different location from the other CUDA DLLs. Ensure that you add the directory where you installed the cuDNN DLL to your %PATH% environment variable.
• GPU card with CUDA Compute Capability 3.0 or higher. See NVIDIA documentation for a list of supported GPU cards.
If you have a different version of one of the preceding packages, please change to the specified versions. In particular, the cuDNN version must match exactly: TensorFlow will not load if it cannot find cuDNN64_6.dll. To use a different version of cuDNN, you must build from source.
“

Many Thanks for the assistance so far,

Graham



Sent from Mail for Windows 10

**Comment by jjallaire** [comment]
I don't really have additional instructions over and above what's on the TensorFlow site. Configuring CUDA for Windows can indeed be a huge pain. This diagnostic script might help: https://gist.github.com/mrry/ee5dbcfdd045fa48a27d56664411d41c

**Comment by gbortz27** [comment]
Thanks

will try

Graham
